### PR TITLE
test: Add subscriber plugin Tests

### LINKED ISSUE #584: Simulator integration tests for closed-range historic streaming
* Add integration tests using the Simulator to verify the API functionality
* Verify the design outlined here: https://github.com/hashgraph/hedera-block-node/pull/577

### LINKED ISSUE #933: Add Consumer Plugin Tests
### Story Form

As a Block Node Developer
I want to have enough code coverage on the Stream Subscriber Plugin
So that I can confidently use and deploy the application in production environments.

### Technical Notes

This effort should focus on improving unit tests for the plugin created under issue #1051.

### PR DESCRIPTION
**Pull request by georgi-l95**
## Reviewer Notes
This PR aim to:
- Introduce minor improvements to object checks, allocations and more.
- Add unit tests for config and session objects, with enough coverage and cases to ensure that we have adequte confidence in the plugin.
- Add E2E covering both negative and positive scenarios for single consumer cases and multiple consumer case.
- Fix issue, where we don't close the session and stop the streaming, when we execute succesfully the request.
- Enable new plugin as default.

Things to consider:
- One unit test disabled due to unstable behavior. Passes when run as standalone, fails when run with all others from the same class.
- Added check in method `allRequestedBlocksSent()` to ensure that when consumer requests 0 as end block, we continue to stream indefinitely.
- Added `close()` session method call in `call()`, when we execute the request, in cases where it's valid and consumer does not want live streaming(0 as `endblock`).

## Related Issue(s)
Fixes #933 
Fixes #584

### CODE REVIEW

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java [hunk 1]
```diff
@@ -111,18 +112,20 @@ public class BlockStreamSubscriberSession implements Callable<BlockStreamSubscri
...
-            SubscribeStreamRequest request,
-            Pipeline<? super SubscribeStreamResponseUnparsed> responsePipeline,
-            BlockNodeContext context,
-            final CountDownLatch sessionReadyLatch) {
+            final long clientId,
+            @NonNull final SubscribeStreamRequest request,
+            @NonNull final Pipeline<? super SubscribeStreamResponseUnparsed> responsePipeline,
+            @NonNull final BlockNodeContext context,
+            @NonNull final CountDownLatch sessionReadyLatch) {
+        requireNonNull(request);
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java:120]: This line isn't needed, the very first use of request will throw if request is null.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java [hunk 2]
```diff
@@ -111,18 +112,20 @@ public class BlockStreamSubscriberSession implements Callable<BlockStreamSubscri
...
+
         LOGGER.log(Level.TRACE, request.toString());
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java:122]: Should probably remove this log line, as it's pretty noisy.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java [hunk 3]
```diff
@@ -174,6 +177,10 @@ public BlockStreamSubscriberSession call() {
                     // then process live blocks, if available.
                     sendLiveBlocksIfAvailable();
                 }
+                if (!interruptedStream.get()) {
+                    // We've sent all request blocks. Therefore, we close, according to the protocol.
+                    close(SubscribeStreamResponse.Code.READ_STREAM_SUCCESS);
+                }
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java:183]: This is not correct, interrupted does not imply we succeeded, it's an interrupt, not all blocks sent.
The close is correct, but only because we don't have an "interrupted" response code.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java [hunk 4]
```diff
@@ -251,6 +259,9 @@ private void awaitNewLiveEntries() {
     }
 
     private boolean allRequestedBlocksSent() {
+        if (endBlockNumber == 0L) { // We are requesting block indefinitely.
+            return false;
+        }
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java:264]: This is not correct.  `0L` as end block is valid if I wish to request block `0`.
Infinite is `UNKNOWN_BLOCK_NUMBER` and is handled in the next line.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java [hunk 5]
```diff
@@ -532,7 +542,7 @@ private void sendOneBlockItemSet(final List<BlockItemUnparsed> blockItems) {
     //       This means we must not modify any state in the session object directly.
     //       Instead we must use a transfer queue to pass the block items to the session
     //       and possibly set an atomic flag if we are "too far behind".
-    private static class LiveBlockHandler implements NoBackPressureBlockItemHandler {
+    static class LiveBlockHandler implements NoBackPressureBlockItemHandler {
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java:545]: This was private deliberately, as we shouldn't be trying to test this implementation detail or exposing it for testing.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java [hunk 6]
```diff
@@ -557,7 +567,7 @@ public void onTooFarBehindError() {
         }
 
         @Override
-        public void handleBlockItemsReceived(final BlockItems blockItems) {
+        public void handleBlockItemsReceived(@NonNull final BlockItems blockItems) {
             if (blockItems.newBlockNumber() > latestLiveStreamBlock.get()) {
                 latestLiveStreamBlock.set(blockItems.newBlockNumber());
                 LOGGER.log(Level.TRACE, "Updated latest block to {0}.", latestLiveStreamBlock);
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java:573]: Probably should remove most of these `TRACE` logs.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java [hunk 7]
```diff
@@ -31,7 +33,22 @@
...
+/**
+ * Provides implementation for the block stream subscriber endpoints of the server. These handle incoming requests for block
+ * stream from consumers.
+ *
+ * <p>This plugin is responsible for:
+ * <ul>
+ *   <li>Managing subscriber sessions for clients requesting block streams</li>
+ *   <li>Handling both live-streaming and historical block data requests</li>
+ *   <li>Implementing the gRPC service interface for block stream subscriptions</li>
+ *   <li>Coordinating between multiple publishers when they connect to the block node</li>
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java:45]: This is an hallucination; this plugin has nothing to do with publishers.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java [hunk 8]
```diff
@@ -31,7 +33,22 @@
...
+ *   <li>Maintaining metrics about active subscribers and stream health</li>
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java:46]: This isn't quite true, stream health isn't really monitored and probably cannot be; it's a nebulous concept at best.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java [hunk 9]
```diff
@@ -157,25 +176,38 @@ Map<Long, BlockStreamSubscriberSession> getOpenSessions() {
...
+    /**
+     * Handler for block stream subscription requests from clients.
+     * Responsible for:
+     * <ul>
+     *   <li>Managing client sessions for block stream subscribers</li>
+     *   <li>Processing subscription requests and streaming block data to clients</li>
+     *   <li>Tracking active subscriber sessions and their lifecycle</li>
+     *   <li>Handling graceful termination of subscriber connections</li>
+     *   <li>Reporting metrics about active subscribers</li>
+     * </ul>
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java:188]: This is highly misleading. Most of these "responsibilities" are _mostly_ handled by the `Session` thread, this class really just supplies the methods to initiate a session and manage the _futures_ for those sessions.
I'd recommend removing "responsible for" sections entirely (particularly as copilot generally gets it completely wrong when generating these).
Perhaps focus on a more prosaic description of the class initially.

**Review by jsync-swirlds — COMMENTED** [review]
Quick initial look at the non-test classes, will look at more on Monday.
A few incorrect documentation items, and a couple over-zealous adjustments.

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java [hunk 10]
```diff
@@ -0,0 +1,398 @@
...
+    private CountDownLatch sessionReadyLatch;
+    private static final long CLIENT_ID = 0L;
+
+    @BeforeEach
+    void setUp() {
+        sessionReadyLatch = new CountDownLatch(1);
+        final Configuration configuration = ConfigurationBuilder.create()
+                .withConfigDataType(SubscriberConfig.class)
+                .build();
+        responsePipeline = spy(new ResponsePipeline());
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java:72]: Using a `spy` to assert invocation counts is concerning; we end up testing far too specifically in almost all cases. I think we'd be better off validating the _result_ of a pipeline rather than trying to count method invocations (particularly as it's entirely reasonable to rework the pipeline quite dramatically but tests for the Session shouldn't fail as a result).

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java [hunk 11]
```diff
@@ -0,0 +1,398 @@
...
+            boolean historicalProvided = false;
+            while (!historicalProvided) {
+                try {
+                    verify(responsePipeline, atLeast((int) MAX_AVAILABLE_BLOCK))
+                            .onNext(any(SubscribeStreamResponseUnparsed.class));
+                    historicalProvided = true;
+                } catch (TooFewActualInvocations e) {
+                    Thread.sleep(100);
+                }
+            }
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java:137]: Would be good to check the responses sent into a buffer, rather than checking a spy method was called.

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java [hunk 12]
```diff
@@ -0,0 +1,398 @@
...
+                        .blockItems(blockItems.blockItems())
+                        .build();
+
+                lenient().when(blockAccessor.blockUnparsed()).thenReturn(blockUnparsed);
+                lenient().when(context.historicalBlockProvider().block(i)).thenReturn(blockAccessor);
+            }
+
+            // Start session in separate thread
+            Thread sessionThread = new Thread(session::call);
+            sessionThread.start();
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java:125]: After start, it might be good to check that the "session ready" countdown latch is unlocked before continuing (that's partially why it exists).

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java [hunk 13]
```diff
@@ -0,0 +1,398 @@
...
+                final SubscribeStreamResponseUnparsed.Builder response = SubscribeStreamResponseUnparsed.newBuilder()
+                        .blockItems(BlockItemSetUnparsed.newBuilder().blockItems(blockItems.blockItems()));
+                Thread.sleep(100);
+                verify(responsePipeline, times(1)).onNext(response.build());
+            }
+
+            // Verify final pipeline state
+            verify(responsePipeline, times(1)).onComplete();
+            verify(responsePipeline, never()).onError(any(Throwable.class));
+        }
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java:153]: Here again, it would be nice to check the actual responses from the pipeline rather than spying and checking invocations...

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java [hunk 14]
```diff
@@ -0,0 +1,398 @@
...
+    @Nested
+    @DisplayName("Input Validation Tests")
+    class ValidationTests {
+
+        /**
+         * Tests the handling of a start block number that is neither available in history
+         * nor in the live stream range.
+         */
+        @Test
+        @DisplayName("Should reject request with start block outside available range")
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java:215]: A lot of these duplicate tests in the plugin tests. Would it, perhaps, be better to focus on testing things here that those tests do not cover?

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java [hunk 15]
```diff
@@ -0,0 +1,398 @@
...
+     * Creates a sample block items set for testing.
+     *
+     * @param blockNumber the block number to create items for
+     * @return a BlockItems object containing sample block items
+     */
+    private BlockItems createSampleBlockItems(long blockNumber) {
+        BlockItemUnparsed blockHeader = toBlockItemUnparsed(sampleBlockHeader(blockNumber));
+        BlockItemUnparsed roundHeader = toBlockItemUnparsed(sampleRoundHeader(blockNumber));
+        BlockItemUnparsed blockProof = toBlockItemUnparsed(sampleBlockProof(blockNumber));
+        return new BlockItems(List.of(blockHeader, roundHeader, blockProof), blockNumber);
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java:342]: We have a fixture method that does this.  Might be good to use that.

It could also be helpful to extend the fixture to include batches that are not complete blocks.  This might enable tests (particularly for live blocks) to verify that the code (using a fixture history provider instead of a mock so history assembles the full blocks) handles a live stream of blocks split across multiple batches (perhaps including several different items, beyond just round header).

Perhaps something to add in the next PR...

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java [hunk 16]
```diff
@@ -0,0 +1,398 @@
...
+     * @param maxBlock the maximum available block number
+     */
+    private void setupHistoricalBlockProvider(long minBlock, long maxBlock) {
+        final BlockRangeSet availableBlocks = mock(BlockRangeSet.class);
+        when(availableBlocks.min()).thenReturn(minBlock);
+        when(availableBlocks.max()).thenReturn(maxBlock);
+        when(context.historicalBlockProvider()).thenReturn(historicalBlockFacility);
+        when(context.blockMessaging()).thenReturn(blockMessagingFacility);
+        when(context.historicalBlockProvider().availableBlocks()).thenReturn(availableBlocks);
+    }
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java:358]: Instead of a mock, this could use the fixture in-memory history provider and just feed in the blocks to be returned...

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java [hunk 17]
```diff
@@ -0,0 +1,398 @@
...
+        return SubscribeStreamRequest.newBuilder()
+                .startBlockNumber(startNumber)
+                .endBlockNumber(endNumber)
+                .build();
+    }
+
+    /**
+     * A simple implementation of Pipeline for testing purposes.
+     */
+    private class ResponsePipeline implements Pipeline<SubscribeStreamResponseUnparsed> {
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java:377]: There is a similar implementation in fixtures (although it's buried and an anonymous inner class) perhaps we could extract that to a top level class to make it easier to use here instead of duplicating it?
```java
        new Pipeline<>() {
            @Override
            public void clientEndStreamReceived() {
                LOGGER.log(TRACE, "clientEndStreamReceived");
            }

            @Override
            public void onNext(Bytes item) throws RuntimeException {
                fromPluginBytes.add(item);
            }

            @Override
            public void onSubscribe(Subscription subscription) {
                LOGGER.log(TRACE, "onSubscribe");
            }

            @Override
            public void onError(Throwable throwable) {
                fail("onError: ", throwable);
            }

            @Override
            public void onComplete() {
                LOGGER.log(TRACE, "onComplete");
            }
        };
```

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/SubscriberConfigTest.java [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+import com.swirlds.config.api.Configuration;
+import com.swirlds.config.api.ConfigurationBuilder;
+import com.swirlds.config.api.validation.ConfigViolationException;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Unit tests for the {@link SubscriberConfig} class.
+ */
+public class SubscriberConfigTest {
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/SubscriberConfigTest.java:15]: Why are we testing configuration?
It seems awfully low value, and also extremely brittle.  I should be able to modify configuration values and defaults without also having to modify test classes.

Note, I know this is different from how we've done things in the past, but one of the things discussed at Devcon was that we should not have tests so tightly coupled to implementation that refactoring is made _more difficult_.  Part of the value of unit tests is to provide confidence when refactoring (because unmodified tests still pass), not add extra work and complexity.

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/SubscriberTest.java [hunk 19]
```diff
@@ -225,6 +225,7 @@ void testSubscriberBlockStreamInMiddleNoHistory() throws ParseException {
         activePlugin.stop(); // request the plugin to end all client streams.
     }
 
+    @Disabled("Disabled until issue with it is resolved. Works when run standalone")
```
Comment (jsync-swirlds) [inline block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/SubscriberTest.java:228]: hmmm... This _was_ working, but it was also one of the hardest tests to get working correctly.
Perhaps send me a DM with what's not working or how it's failing?  I'd like to understand this better.

File: suites/src/main/java/org/hiero/block/suites/subscriber/positive/PositiveSingleSubscriberTests.java [hunk 20]
```diff
@@ -0,0 +1,173 @@
...
+            } catch (InterruptedException e) {
+                // Do nothing, this is not mandatory, we try to shut down cleaner and graceful
+            }
+        });
+        simulators.forEach(simulator -> simulator.cancel(true));
+        simulators.clear();
+    }
+
+    @Test
+    @DisplayName("Should subscribe to stream and receive both historical and live")
```
Comment (jsync-swirlds) [inline suites/src/main/java/org/hiero/block/suites/subscriber/positive/PositiveSingleSubscriberTests.java:53]: Just a note here.  It's almost impossible (from outside the node) to know if you received a block from history or live.  The subscriber is intended to conceal the source as effectively as possible so that the subscriber just receives the stream, regardless of live or historical.
For this test, all we really can know for certain is that we receive the stream timely; the effective source of each block is pretty obscured.

**Review by jsync-swirlds — COMMENTED** [review]
Read through the rest.
Some concerns with use of mocks when fixtures (which already exist) might be better.
Also a few other small suggestions or questions.

Thanks for writing the Suite tests, those help a lot with confidence that everything works together the way it should.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java (hunk 4 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java]: My understanding was the same (-1 is indefinetely only). However then I saw the protocol definitions, [which](https://github.com/hashgraph/hedera-protobufs/blob/main/block/block_service.proto#L589-L609) states that: 
```
     * This field MAY be zero (`0`) to indicate the stream SHOULD continue
     * indefinitely, streaming new blocks as each becomes available.<br/>
     * If this value is greater than zero (`0`)
```

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java (hunk 7 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java:45]: I was copying and pasting some templates for the javadoc around and didn't edit them correctly. My bad, thank you

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java (hunk 8 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java:46]: You're right. Will remove.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java (hunk 1 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java]: Thank you for resolving this. 🙏

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java (hunk 2 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java]: Thank you for resolving this. 🙏

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java (hunk 3 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java]: Thank you for resolving this. 🙏

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java (hunk 4 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java]: I also think that -1 should be the only case for indefinitely streaming. That means we need to change the documentation in the protobufs.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java (hunk 5 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java]: Yes, I was trying to test it out. Thank you for resolving this.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java (hunk 6 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java:579]: Removed.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java (hunk 9 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java:188]: Edited

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java (hunk 8 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java:46]: Removed.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java [hunk 21]
```diff
@@ -41,8 +58,9 @@ public class SubscriberServicePlugin implements BlockNodePlugin, ServiceInterfac
     private final List<SubscribeBlockStreamHandler> clientHandlers = new LinkedList<>();
     /** The block node context */
     private BlockNodeContext context;
-
+    /** The subscriber plugin configuration */
     private SubscriberConfig pluginConfiguration;
+    /** The client handler for client subscriptions*/
```
[the remark this answers is not in the document]
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java:63]: Gotcha, edited. Thanks

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java [hunk 22]
```diff
@@ -51,8 +69,9 @@ public class SubscriberServicePlugin implements BlockNodePlugin, ServiceInterfac
      * {@inheritDoc}
      */
     @Override
-    public void init(BlockNodeContext context, ServiceBuilder serviceBuilder) {
-        this.context = context;
+    public void init(@NonNull final BlockNodeContext context, @NonNull final ServiceBuilder serviceBuilder) {
+        requireNonNull(serviceBuilder);
```
[the remark this answers is not in the document]
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java:73]: Removed. You're right.

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java [hunk 23]
```diff
@@ -157,25 +176,38 @@ Map<Long, BlockStreamSubscriberSession> getOpenSessions() {
...
         private final Logger LOGGER = System.getLogger(getClass().getName());
         /** Count of active sessions, because LongGauge doesn't support increment/decrement */
-        private AtomicLong sessionCount = new AtomicLong(0L);
+        private final AtomicLong sessionCount = new AtomicLong(0L);
         /** The next client id to use when a new client session is created */
         private final AtomicLong nextClientId = new AtomicLong(0);
-
+        /** The block node context */
         private final BlockNodeContext context;
+        /** The subscriber plugin responsible for handling incoming requests*/
```
[the remark this answers is not in the document]
Comment (georgi-l95) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/SubscriberServicePlugin.java:199]: Edited. thanks

File: suites/src/main/java/org/hiero/block/suites/subscriber/positive/PositiveSingleSubscriberTests.java (hunk 20 above)
Comment (georgi-l95) [reply suites/src/main/java/org/hiero/block/suites/subscriber/positive/PositiveSingleSubscriberTests.java:53]: Yes, in real scenario we wouldn't know. Now we can determine now due to the simulator. You're right.

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java (hunk 17 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java]: I can open an issue for us to adress that. I think there is a similar pipeline in the publisher tests also, which could be addressed in the said issue 🤔

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java (hunk 16 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java]: Thank you for adressing this.

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java (hunk 14 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java]: Agree, wanted to make sure we cover some edge cases, but you're right.

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java (hunk 12 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java]: Agree with you 👍  Thank you for addressing this

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java (hunk 11 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java]: Yes, thanks for adding addition lists and getters for us to check this in the pipeline class.

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java [hunk 24]
```diff
@@ -0,0 +1,398 @@
...
+    private BlockNodeContext context;
+
+    @Mock
+    private Metrics metrics;
+
+    @Mock
+    private HistoricalBlockFacility historicalBlockFacility;
+
+    @Mock
+    private BlockMessagingFacility blockMessagingFacility;
```
[the remark this answers is not in the document]
Comment (georgi-l95) [reply block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java]: Thanks for addressing this.

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java (hunk 15 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSessionTest.java]: Good idea, opened a issue https://github.com/hiero-ledger/hiero-block-node/issues/1086

File: block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java (hunk 4 above)
Comment (jsync-swirlds) [reply block-node/stream-subscriber/src/main/java/org/hiero/block/node/stream/subscriber/BlockStreamSubscriberSession.java]: The proto documentation is definitely wrong.  I keep saying I should make a clean pass to ensure it's all "correct"; time just hasn't been on our side in that regard.

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/SubscriberConfigTest.java (hunk 18 above)
Comment (jsync-swirlds) [reply block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/SubscriberConfigTest.java:15]: I would lean toward just not having testing for config classes directly; any "does this work correctly with incorrect values" should probably be in the unit tests for the code _consuming_ the config, rather than the config class.

File: suites/src/main/java/org/hiero/block/suites/subscriber/positive/PositiveSingleSubscriberTests.java (hunk 20 above)
Comment (jsync-swirlds) [reply suites/src/main/java/org/hiero/block/suites/subscriber/positive/PositiveSingleSubscriberTests.java:53]: I had not considered that the simulator could be used to ensure the node reads from one source or another; a useful feature.

**Review by jsync-swirlds — DISMISSED** [review]
Looks good.  Thanks for working through this.

**Review by jsync-swirlds — APPROVED** [review]

File: block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/SubscriberConfigTest.java (hunk 18 above)
Comment (georgi-l95) [reply block-node/stream-subscriber/src/test/java/org/hiero/block/node/stream/subscriber/SubscriberConfigTest.java:15]: Opened a issue to remove this class and others similar (publisher also has). So that all can be addressed in another effort and we don't get out of scope for this one. https://github.com/hiero-ledger/hiero-block-node/issues/1090
